class.jetpack.php

```diff
@@ -4519,7 +4519,7 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 		}
 
 		if ( $from ) {
-			$url = add_query_arg( 'from', $from, $url );
+			$url = add_query_arg( 'from', sanitize_key( $from ), $url );
```

Comment:
I just recalled why we didn't do this before. There are several instances where we have a `.` in the `from` argument, for example `jpo-1.6.1-bluehost`. Such instances would have the dot removed, which I guess isn't expected. cc @Automattic/catalyst if they know more if we'd need those in the future.

Reply:
I suppose we could use `esc_attr()` instead for now, to avoid that issue? I am happy to make the change.

Reply:
Well, the thing is we shouldn't be escaping at this time, as it's a bit early. 

But we could totally use `sanitize_text_field()` which won't strip out any wanted characters, but will still remove malicious stuff.

Reply:
Done in 0ffa0a14cb95dd92eefb6d0440e9edd55885c52f